tests/pytests/unit/states/test_zabbix_template.py

```diff
@@ -0,0 +1,329 @@
+        defined_obj,
+        [],
+        defined_c_list_subs["applications"],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+    ]
+
+
+@pytest.fixture
+def existing_obj():
+    return [
+        {
+            "available": "0",
+            "tls_connect": "1",
+            "maintenance_type": "0",
+            "groups": [{"groupid": "1"}],
+            "macros": [
+                {
+                    "macro": "{$CEPH_CLUSTER_NAME}",
+                    "hostmacroid": "60",
+                    "hostid": "10206",
+                    "value": "ceph",
+                }
+            ],
+            "hosts": [{"hostid": "10112"}, {"hostid": "10113"}],
+            "status": "3",
+            "description": "Template for Ceph nodes",
+            "host": "A Testing Template",
+            "disable_until": "0",
+            "templateid": "10206",
+            "name": "A Testing Template",
+        }
+    ]
+
+
+@pytest.fixture
+def substitute_params_exists(defined_obj, existing_obj):
+    return [
+        defined_obj,
+        existing_obj[0],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+    ]
+
+
+@pytest.fixture
+def existing_obj_diff():
+    return [
+        {
+            "groups": [{"groupid": "1"}],
+            "macros": [
+                {
+                    "macro": "{$CEPH_CLUSTER_NAME}",
+                    "hostmacroid": "60",
+                    "hostid": "10206",
+                    "value": "ceph",
+                }
+            ],
+            "hosts": [{"hostid": "10112"}, {"hostid": "10113"}],
+            "status": "3",
+            "templateid": "10206",
+            "name": "A Testing Template",
+        }
+    ]
+
+
+@pytest.fixture
+def substitute_params_update(defined_obj, existing_obj_diff):
+    return [
+        defined_obj,
+        existing_obj_diff[0],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+    ]
+
+
+@pytest.fixture
+def diff_params():
+    return {"old": {}, "new": {"macros": [], "templateid": "10206"}}
+
+
+@pytest.fixture
+def configure_loader_modules():
+    return {zabbix_template: {}}
+
+
+def test_present_create(substitute_params_create):
+    """
+    Test to ensure that named template is created
+    """
+    with patch("salt.states.zabbix_template.CHANGE_STACK", []):
+        name = "A Testing Template"
+        ret = {"name": name, "result": False, "comment": "", "changes": {}}
+
+        def side_effect_run_query(*args):
+            """
+            Differentiate between __salt__ exec module function calls with different parameters.
+            """
+            if args[0] in ("template.get", "application.get"):
+                return []
+            elif args[0] == "template.create":
+                return {"templateids": ["10206"]}
+            elif args[0] == "application.get":
+                return []
```

Comment:
```suggestion
```
We are now covering this branch in line 154

Reply:
Fixed